### ISSUE TITLE
Enabling Gutenberg in WP categories

### ISSUE THREAD

**Issue by FerrielMelarpis** [WordPress/gutenberg#17099]
Related thread: https://wordpress.org/support/topic/gutenberg-category-editor/

Not sure if this is possible now but I haven't seen any documentation regarding this.

Is WP category going to be supported by Gutenberg as well?

Right now, for our use case, we depend on ACF to contain the data needed for representing the structure for our category pages. It would be nice to have a way to represent the category data as blocks as well.

**Comment by TenentePM** [comment]
Same here, I'd like to edit category-descriptions with Gutenberg.

**Comment by Flaschenzug** [comment]
I'd like that too! Wordpress has taken a big step with Gutenberg. But this step is only meaningful if it is also consistent. Why shouldn't the categories be handled in the same way as posts and pages?

**Comment by mtias** [comment]
This would be a cool idea

**Comment by danyork** [comment]
Just noting that several of us left comments expressing our support for this task over in the now-closed  #12511 . 

For my employer's main site, we use category archives pages extensively in the navigation and main design. They are currently built them out using ACF with rows of text, images, etc. above the posts for the category (or tag). We've been doing more and more with the block editor across our site and building some excellent pages. We really want to now use some of the Gutenberg blocks and techniques on the Category pages as well (and also Tag archive pages - but our priority is on the Category pages).

Unfortunately, I don't have the programming expertise to help with the work, but I would be glad to help with whatever testing is needed!

**Comment by rpldom** [comment]
We use Woocommerce category pages as a main landing pages in our store. This structure is pretty common for many stores. And abundance of Gutenberg editor on category pages really restricts further development of our site, I think category and taxonomy pages should be treated as posts and pages and have similar functionality to them.

Also when I want to add a link Gutenberg isn't showing Woocomerce category pages as available option.

**Comment by paaljoachim** [comment]
This post, by @gziolo  mentions new filters/hooks that seems to make easier to blockify the WP categories page. (Recreate it with Gutenberg.) https://make.wordpress.org/core/2021/06/16/block-editor-api-changes-to-support-multiple-admin-screens-in-wp-5-8/

@critterverse @javierarce 
Perhaps this would be an interesting screen to design.

**Comment by gziolo** [comment]
@paaljoachim, as explained in my response to your comment on a linked post, it's a more complex process to enable the block editor on a new screen that using a few API methods described in the dev note. The widgets screen might be a good way to think about it. It took a lot of effort to get the block paradigm integrated there.

**Comment by paaljoachim** [comment]
Thank you Greg!

**Comment by skorasaurus** [comment]
With the increased focus and transition to full site editing; this shouldn't be forgotten :) 

https://github.com/WordPress/gutenberg/issues/37746 has a clear use case that reminds me some retail/woocommerce shops use categories extensively.

**Additional use case:** 
I [have a site with couple different podcasts](https://cpl.org/podcasts/) (podcast is the custom post type, podcast_series is the taxonomy each new post is a new 'podcase episode'). 

I want a landing/index page for each podcast series that we have: enabling Gutenberg for category or custom taxonomy pages would enable me to build this. 

With the lack of native Gutenberg support; I worked around it by: 

- register a widget for each term that you want to use with `register_sidebar` (as an example - https://developer.wordpress.org/reference/functions/register_sidebar/#comment-3580) 

- creating php template (taxonomy-podcast_series.php) for the custom taxonomy, where I
      -  write a function to get the first term for the given taxonomy 
      -  based on which term is used on the page; display a certain widget's contents which acts the 'landing page'. 

``` * @param  int    $post_id  Post ID
 * @param  string $taxonomy Taxonomy slug
 * @return WP_Term|false
 */
function get_first_term( int $post_id, string $taxonomy ) {
	$terms = get_the_terms( $post_id, $taxonomy );
	if ( empty( $terms || is_wp_error( $terms ) ) ) {
		return false;
	} else {
		return reset( $terms );
	}
}

$first_term = get_first_term( get_the_ID(), 'slug_of_your_taxonomy' );

if ($first_term->name === 'Name of Podcast A') { 
dynamic_sidebar( 'name-of-the-widget-that-you-made-for-podcast-a );
} 

```




(You could also `<?php echo do_blocks( '<!-- block markup /-->' ); ?>`)

**Comment by gingerling** [comment]
Hi, now that there are templates for category and product category pages coming out, this issue becomes more important I think.

In my view there is no "full site editing" without this vital feature.

I can totally see people doing hacky things like making a separate template for each category page, for example.

How to we get this issue on the roadmap? I have no idea of the scale of this task. Could we team up and try and fund some development on this? Seems like a fair few WooCommerce stores on here!

**Comment by only-sadeghi** [comment]
is there a way to do it?🥺😐

**Comment by gingerling** [comment]
> is there a way to do it?pleading_faceneutral_face

Hi, this https://en-gb.wordpress.org/plugins/visual-term-description-editor/ gives a basic editor, but not a block editor. Perhaps this in combination with the new theme editor for category archives will allow a better experience while this issue is worked on 

Please, if you have other suggested plugins add them while it's not a support forum I think it's helpful for us to assess the need and current capabilities

**Comment by sangtlee** [comment]
We would also love to see block editing enabled on term edit screens.  It's not uncommon to need block editing features on entities that need to function as taxonomy terms.  This is our current workaround to accomplish this:

- Custom post type called "Industry" with rewrite set to false << all of the "block" content is managed on these posts
- Custom taxonomy called "Industries" with slug of "industries" << used for usual taxonomy tagging, querying, etc
--  Taxonomy has a ACF post object field ("parent_post") that lets editors associate this term with a specific "industry" post
- Content items across the site can then be tagged with an Industries term, but when the term archive page is requested, we grab the Block content from the associated page and inject that into the archive page template above the normal loop output << this task is rather trivial with Timber/Twig:
```
{% set parent_post = Post(post.parent_post) %}
  {{ parent_post.content }}
```
- Some additional rewrites and customs are needed to ensure users don't end up on the CPT post, breadcrumbs, etc.

It's not elegant but it works...new editors get confused between the "Industry" CPT and the "Industries" taxonomy...and often they forget about that all-important "parent_post" field on the term.

**Comment by justwheel** [comment]
Another +1 for this feature… the use case I had was to add a Jetpack newsletter subscription widget at the top of a category. Personalizing the category page is an important part of how we use WordPress.

CC: @vansika

**Comment by MrVibe** [comment]
How can WordPress be upgraded to FSE without supporting this feature ?

**Comment by klockfeber** [comment]
This would be very useful for our category pages as well, see an example of how the content looks now when it is only text [here](https://klockfeber.se/seiko/). I am surprised that WordPress 6.0 was released without Gutenberg on categories.

**Comment by Jabe64** [comment]
+1 I created an issue about it 4 years ago:
[https://github.com/WordPress/gutenberg/issues/12511](https://github.com/WordPress/gutenberg/issues/12511)

The issue was closed because Gutenberg was still in early stage. Full Site Editing improves but still no news about this.

Every update I hope to see something then my hopes varnish 😢

@mtias Is it still considered or will categories description be abandoned and should we use FSE template per category instead? 🤔

**Comment by djcowan** [comment]
This is an interesting thread. I have always assumed the non inclusion of advanced term features was due to the original implementation of the taxonomy system. Given Wordpress' evolution: Posts, Tags => non heirachical, transient to the addition of Pages, Categories = Heirachical, static; it makes sense -to me- that a universal one-size-fits-all approach would entail an extensive rewrite of a lot of core functionality.
One of the greatest appeals of the Wordpress system has always been Wp's adherance to compatibility, usability and accessability. Where a number of -other- systems continue to add features regarless of bloat, backward compatibility or viable upgrade path, the Wordpress team and contributors consider every user of the ecosystem at every stage of the development cycle.
So, yes; additional taxonomy and term features would be nice for some -me included- but they are not essential to the core functionality of Wordpress. As Wordpress evolves, perhaps these enhancements can be made; perhaps not.
There are serveral alternatives. 
1. See this as an opportunity to build a plugin to enhance the system.
2. Join the contribution team and assist in developing Wordpress
3. Adjust your approach to working with the system as it currently is.
4. Hire a development team to build the functionality you can't live without.
Thank you for listening: Please like and subscribe to my soapbox for more uniformative and unwanted opionated comments.

**Comment by jak-kal** [comment]
Any movement on this? Was about to finally take the plunge with Gutenberg but this is a serious drawback! I really don't want to use Gutenberg on pages and ACF repeaters on WooCommerce categories.

The only feasible way around this that I can see is to create a corresponding page for each landing category but that's going to really confuse other users.

**Comment by jonathan-dejong** [comment]
I've been thinking about this. I agree with the general consensus that this would be very powerful for many use cases. Especially for stores that use shop categories as landing pages which is extremely common. 

Even with FSE in the state it is today it has one huge drawback, there is no context. 
I can create a Block Template part to show on location N on each term archive, but the content within does not adjust to the specific term that is being displayed. 

As to the claim that this requires a whole lot of rewrite I'm not so sure it does? 
The `wp_terms` table currently has `term_id`, `name`, `slug` and `term_group`. 
What if we were to add a `term_content` column which would act the same way `post_content` does for posts? 
_sidenote: Lets please not stick this into termmeta, making it utterly inefficient_ 

We could add a new parameter to `register_taxonomy()` to opt in on the block editor , something like `enable_block_editor` OR maybe even consider some more consolidation with how we register CPTs and future proofing by adding a `supports` parameter which can take several future parameters.
_Sidenote: I wish CPT registration had block_editor as a parameter here as well, but that's a different topic._

So it would be fully backwards compatible. Very unlikely to clash with any existing plugins. 
Custom registered fields can be treated the same way they are on block post types, as "meta" added to the bottom of the page. 
Core fields like "name" and "slug" becomes the title and permalink same way as with posts. 
"Description" becomes a paragraph block (if one exists). 

I'm sure this'll be met by some as "make it a plugin" and sure I don't disagree with that.. But then make it part of the core feature plugin https://wordpress.org/plugins/gutenberg/

That's my 🪙 🪙

**Comment by aurooba** [comment]
Just popping in here to add my support for this (literally because I'm about to dive into a project that would be simplified if this was already a thing) and I like the approach suggested by @jonathan-dejong. :)

**Comment by goaround** [comment]
I think a good place would be the https://github.com/Automattic/blocks-everywhere Plugin. I've created a request some time ago: https://github.com/Automattic/blocks-everywhere/issues/20 but no progress for now.

**Comment by eric-michel** [comment]
I've run up against this issue several times as well. It seems like extending Gutenberg support to term descriptions would be integral to the Full Site Editing project. But with the recent announcement that phase 3 is the next big focus going forward, that implies that Full Site Editing improvements are going on the back burner. I'm worried term descriptions will just get left behind, especially given how old this issue is.

**Comment by michael-sumner** [comment]
I too am also expressing support for the Gutenberg within taxonomy 👋

We have a set of taxonomy templates which must contain particular blocks for each taxonomy.

However, as it is not baked into WP Core, we are attempting to make use of other methods to create the templates for the taxonomy.

Furthermore, we are specifying the template in the back-end, so that we can make use of "available templates" to work from.

This stops the use of "hard-coded slugs" in the WP template hierarchy.

Of course, we can cache the request made to specify the template.

**Comment by alex-t445** [comment]
Hello, from the perspective of a user on this subject:  

I resisted Gutenberg for years, something that I regretted later. Today Classic Editor seems so clunky and out-of-date. To me, having to use it, it's like telling an adult to get on a tricycle again.  

I switched my site to an FSE theme a few days ago. I really had to twist the arm of the young man who does coding for me to learn FSE. He is slowly getting it. Over and over, I beseeched him to embrace new technology. Lecture after lecture. 

But then right in the middle of all this talk about blockifying everything: BANG! I run into the fact that the Term Description block can only be edited with Classic. It is so bizarre. Why? Why?  I keep asking. 

**Our site uses categories and terms as landing pages.** There is so much we want to do with  terms and categories, but having to use Classic is like trying to work with one hand tied behind your back. 

I can't emphasize how ridiculous this makes the Wordpress FSE infrastructure look. Please give us Gutenberg and free us from this albatross from the days gone by.

**Comment by michaelbourne** [comment]
I think the system described by @jonathan-dejong makes perfect sense. Huge vote of approval here.

**Comment by AntoinePBorg** [comment]
Ooh - +1 - me too! me too! 

I'm surprised this hasn't been done already given there are so many people who want it (and have wanted it for so long)!

**Comment by MathieuMarteau** [comment]
I also don't understand why this feature is still not present. So many people use taxonomies and categories... It's frustrating not to be able to use our modular blocks on this kind of page. Please do something about it...

**Comment by djcowan** [comment]
I'd rather the community remained focused on refactoring Wordpress' core functionality. Oh, and the incredible work they're doing with accessibility.

**Comment by jonathan-dejong** [comment]
> I'd rather the community remained focused on refactoring Wordpress' core functionality. Oh, and the incredible work they're doing with accessibility.

I don't understand this comment. The community can do more than one thing at once.
Also wouldn't this very much be qualified as "refactoring WordPress core functionality" anyway 🤔

**Comment by paaljoachim** [comment]
The thing is that anyone can spend a bit of time adding wireframes/mockup with suggestions as to how this might look like. That will create some motion. It will naturally need a bit forth and back in regards to design. After a design has been decided upon then a developer can go ahead and create a PR (Pull Request) and so the coding exploration begins. Core designers and developers will need to also give feedback along the way.

**Comment by djcowan** [comment]
Yes, I agree, the community, contributors, and core team can work together on multiple facets of the system.
But when I referred to 'core' functionality, I was thinking more in term of the Wordpress' core DNA rather than the core application code.
Sure, there are many volunteer contributors and there can always be room for more but the project needs to be planned, coordinated, managed and rigorously tested as well. This obviously includes the code, which is a minefield in itself but also requires input and advice from the teams who work on accessibility, design, user experience, marketing, translation and more besides. 
I've personally not contributed so I am not one to complain but @paaljoachim is correct, there are lots of ways for each of us to become involved. I apologise if my comment caused undue offence. It was not intended as such.

**Comment by sangtlee** [comment]
IMHO, taxonomy terms should retain all existing functionality...i.e. name, slug, parent terms, description, tag_id, archives...and layer in the new aspects: block editor, templates, draft/published states, featured image, and author.  A quick mockup of the merged UI might look like the attached.

![term_block_add_mockup](https://github.com/WordPress/gutenberg/assets/913984/e1b9b743-cd54-4213-9144-79a50246c510)

**Comment by jonathan-dejong** [comment]
@djcowan Not at all, I just didn't understand the statement and it didn't really feel like it contributed to the discussion. 
I'm sorry you felt the need to apologise :) 

@sangtlee I like that rough mockup. It makes sense to actually put those fixed fields currently on the term edit screen as fixed fields in the sidebar. 

AFAIK the things to cover would be:
* slug
* Parent
* Description

Slug would be auto-populated by title same as current behaviour. 

I still think it would be nicer to not have description as a fixed field but I recognise that its probably a must to maintain decent backwards compatibility.

**Comment by kobykorman** [comment]
Same as many others mentioned, old-time WP'er here that was a way for a while, and kinda astonished that in 2024 Taxonomies are still not treated like posts/pages by the entire ecosystem. Slugs, revisions, SEO plugins, etc. etc. are all needed and missing, and we're left hacking away with ACF.

**Comment by samunderwood** [comment]
I've noticed a trend where many websites use WordPress's taxonomy system for organising content in the backend, but don't utilise taxonomy term pages on the frontend due to its limited flexibility for content.

They're instead using pages, which are then supplemented with custom blocks to display the post list, alongside all the other custom blocks they want to be able to use on term pages.

I'm still surprised how long Gutenberg has been out and we still are relying on hacky workarounds to use its functionality on something as prevalent on websites as category pages

**Comment by gregdotelphotography** [comment]
This is a must have for Wordpress. I have 5 websites and all of them need this feature. I have to use hacks for it to work properly.

**Comment by asafm7** [comment]
I recently posted a message about it in the core-editor WordPress Slack channel:

https://wordpress.slack.com/archives/C02QB2JS7/p1714163585750229

Feel free to chime in.

**Comment by jonathan-dejong** [comment]
> I recently posted a message about it in the core-editor WordPress Slack channel:
> 
> https://wordpress.slack.com/archives/C02QB2JS7/p1714163585750229
> 
> Feel free to chime in.

Hah okay .. ill buy that the issue title could be clearer but when it was created there wasn't a proposal, just a grievance to discuss. 
The thing is.. my experience is that when I create a new issue to address something that already exists, the issue gets closed immediately referencing that "there is already an issue for this". 

So I'm curious what the process looks like here. 

I wouldn't mind taking the time to create a new structured issue with a clear title provided that the approach I've had in mind is sound. But I would like to know:
1. Is it sound? From someone actually part of the Gutenberg team. 
2. Will it not just be closed as a duplicate? What is the defined process (if any)?

I'm not intentionally trying to come off as a bit sour here, but Ive experienced too many times that my efforts to contribute to this OS projects haven't made the slightest change due to the gatekeepers disinterest. 

For reference here's my proposal (and then there's som discussions around it later in the comments) 
https://github.com/WordPress/gutenberg/issues/17099#issuecomment-1515854204

**Comment by colorful-tones** [comment]
> I recently posted a message about it in the core-editor WordPress Slack channel:
> https://wordpress.slack.com/archives/C02QB2JS7/p1714163585750229
> Feel free to chime in.

That was me suggesting in Slack 👋 

> the issue title could be clearer but when it was created there wasn't a proposal, just a grievance to discuss.

I feel like #37746 does a far better job explaining (with screenshots) what the issue is and the suggested solution. I have the ability to edit this Issue's (17099) description and title, but I feel like it could be misleading for the original creator: @FerrielMelarpis. This was why I suggested we think about creating a new issue with a reference to this issue (17099), and then close it.

For what its worth - I think this overall functionality will eventually be explored in the **Phase 3: WordPress admin redesign #53322**. 👈 I _highly_ recommend folks review the videos and overall conversation and contribute your feedback on #53322. Here are some screenshots (please see below) of areas that _could possibly_  be explored for the category screen (please do not read this and assume it is fact - it is currently being discussed and you should feel empowered to contribute your thoughts) based on @SaxonF original video artifacts in #53322 

There may even be more recent work that is moving closer to the category redesign, but I just could not find it when searching through the many issues. 😄 

![Screenshot 2024-04-29 at 2 21 00 PM](https://github.com/WordPress/gutenberg/assets/405912/9a2796da-161f-4a29-8862-c2eeb2b2d2ab)
![Screenshot 2024-04-29 at 2 22 00 PM](https://github.com/WordPress/gutenberg/assets/405912/f986d6b9-c8a6-43a1-a964-4637cf62e714)

**Comment by asafm7** [comment]
@colorful-tones, I went over the issue you mentioned, but, unless I'm missing something, it doesn't seem to be closely related to the issue of enabling the Block Editor for archive page descriptions. The issue you referred to also feels like an extensive project, making it unlikely to progress significantly soon.

Do you think there is another way to raise and promote the issue presented in this thread?

**Comment by djcowan** [comment]
Might it suffice to add block support to the category description editor. 
Similar to the support/issue reporting field in the Wordpress codex. Perhaps implemented in a similar way to the Freeform core/ block? ![image](https://github.com/user-attachments/assets/4d955fd1-5151-4de7-8dad-e694d393f8a4)

As a postscript. Adding a Category image programmatically has always required a conditional check or adapter for woocommerce, third-party brand plugins etc. it would be nice to standardise the term post meta

**Comment by goaround** [comment]
I submitted a PR https://github.com/Automattic/blocks-everywhere/pull/178 to Blocks Everywhere to enable Gutenberg for the term description. It works quite well with the help of @1ucay. Feel free to give it a try.
As far as I know, Blocks Everywhere is used in bbPress as well.

The only issue is that Blocks Everywhere requires the Gutenberg Plugin. That holds me back from using it in production.

**Comment by asafm7** [comment]
@djcowan @goaround it looks like a good direction.

The Blocks Everywhere plugin has only about 30 active installs though, and hasn't been updated for 10 months. Adding the fact that as @goaround mentioned it requires the Gutenberg plugin which is beta by definition, it doesn't feel like a good solution for production.

Maybe a similar solution can be brought to core as a temporary solution until the admin side gets refactored (which I assume can take a long while).

**Comment by kraftner** [comment]
> Maybe a similar solution can be brought to core as a temporary solution until the admin side gets refactored (which I assume can take a long while).

While I am also desperately waiting for this I have to strongly disagree here. Due to the backwards compatibility commitment of WP nothing ever is temporary. If it takes time that sucks but is still better than getting a half-baked stopgap solution that will only cause much more pain in the long term.

Just as an idea: If you really need something like this a competent developer could probably create a stop-gap solution that e.g. creates a post type that shadows a taxonomy and keeps them in sync. Honestly though if anyone has the time or money to fund that it would probably be better spent in working on the core admin overhaul.

**Comment by asafm7** [comment]
@kraftner that's interesting.

I understand what you are saying, but I'm not sure it applies here.

I feel like moving away from the current admin to the new editor experience is something that is bound to happen, so a temporary fix won't prevent it from happening. It will take time though. I'm not familiar with the process, but my gut feeling is that it can take years (again, just a gut feeling).

> If you really need something like this a competent developer could probably create a stop-gap solution that e.g. creates a post type that shadows a taxonomy and keeps them in sync.

Yes, something like this can be created, but such a workaround also needs to be maintained and is prone to breaking.

Considering that moving away from the current admin will eventually happen, that it will take time though, that many people need this change in the meantime, and that a decent fix has already been found and implemented (just not in a stable way), I don't see a true downside in bringing this change to core.

It can be disabled by default and enabled with a filter or a similar method.

But maybe I'm missing something.

**Comment by kauaicreative** [comment]
On a recent build I needed this functionality. My solution was to create a block that displays related page data in the archive template - I place this block into the archive template. The block checks if a page existed with the same permalink as the archive e.g. services/planning. If there is a match then I insert that page content into the block. This was a nice solution because the client only needs to edit the related page and I can continue to use the term description in the archive template as it was intended. You can see it in action here https://destinationbydesign.com/services/ 

This could just as easily be done with a simple shortcode:

```php
// Place [page-content] anywhere in the archive template
// And make sure that there is a page with the exact same permalink e.g. services/planning

add_shortcode( 'page-content', function ( $atts, $content = '' ) {
  global $wp;
  
  if ( is_tax() ) {
	  $taxonomy_slug = add_query_arg( [], $wp->request ); // e.g. 'services/recreation-master-planning'
	  $page          = get_page_by_path( $taxonomy_slug );
	  if ( isset( $page->post_status ) && $page->post_status === 'publish' ) {
		  $content = do_blocks( $page->post_content );
	  }
  }
  
  return $content;
} );
```

**Comment by WesCook** [comment]
I came up with a similar solution, @kauaicreative.  Instead of modifying a template, I wrote a function to serve a regular page instead of a category page if it matches the same path.  Specifically it looks for matching slugs under a child page `/category-replacements/` to keep things organized.  That page can be hidden or redirected to avoid duplicate content issues.

```php
// Replace any category pages with a regular page of the same slug found under the /category-replacements/ parent
add_action('template_redirect', function() {
	// Do nothing in wp-admin
	if (is_admin()) {
		return $query_vars;
	}

	if (is_product_category()) {
		// Get the current category object
		$current_category = get_queried_object();

		// Get the full URL of the category
		$category_url = get_term_link($current_category);

		// Parse the URL to get the path
		$parsed_url = parse_url($category_url, PHP_URL_PATH);

		// Remove leading and trailing slashes, then replace slashes with hyphens
		$category_path = trim($parsed_url, '/');
		$page_slug = str_replace('/', '-', $category_path);

		// Check if the parent page exists
		$parent_page = get_page_by_path('category-replacements');

		if ($parent_page) {
			// Get the ID of the parent page
			$parent_id = $parent_page->ID;

			// Query for a child page with the constructed slug under the parent page
			$args = array(
				'name'        => $page_slug,
				'post_type'   => 'page',
				'post_parent' => $parent_id,
				'numberposts' => 1
			);
			$child_pages = get_posts($args);

			if (!empty($child_pages)) {
				// Redirect to the child page template
				global $wp_query;
				$wp_query->is_page = true;
				$wp_query->is_single = false;
				$wp_query->is_category = false;
				$wp_query->queried_object_id = $child_pages[0]->ID;
				$wp_query->post = $child_pages[0];
				$wp_query->post_count = 1;
				$wp_query->posts = array($child_pages[0]);

				// Set up the post data
				setup_postdata($child_pages[0]);

				// Use the child page template
				include(get_page_template($child_pages[0]->ID));
				exit;
			}
		}
	}
});
```

For example, if you wanted to replace the Services > Accounting subcategory (`/services/accounting/`), then you'd create a regular page at the path `/category-replacements/services-accounting/`.  The top-level page is always required, and hyphens are used to separate children after that.

It's definitely a hack, but it does work.  Of course I'll dismantle it the very day that WordPress gets native support for Gutenberg in categories.  Especially for cases like WooCommerce that require feature-rich content, the classic text editor just doesn't cut it anymore.

My thanks to anyone working on this problem.  I hope that any potential solution will also be available in classic themes, and not exclusive to block themes.

**Comment by sangtlee** [comment]
Good to see some chatter on this again!  I agree with @kraftner and others that you can achieve what we need with technical workarounds...but IMHO, the real benefit of enabling Gutenberg natively in Term edit pages is improving content editor user experience.  Workarounds like [mine](https://github.com/WordPress/gutenberg/issues/17099#issuecomment-1022876954), using shadow posts or ones using shortcodes inside term description fields are not very intuitive to content editors.

My team also works with Drupal...and almost every Drupal site we've built has at least one Taxonomy (e.g. blog category or degree program) with Terms in those Taxonomies that have block content managed on them which is then displayed on the Term pages...and the content mgmt experience in the Term page is the same as any other block-enabled page.

So what is the typical dev workflow for changes to core?  Can anyone submit PR's?  Or must the work be initiated by a core maintainer?  How can we elevate the visibility of this?  This thread is 5 years old.

**Comment by djcowan** [comment]
> It can be disabled by default and enabled with a filter or a similar method.
___
Would love an opt-in mechanism via theme_support, hook or filter.
The post except field includes the [`excerpt_allowed_wrapper_blocks`] action hook (<https://developer.wordpress.org/reference/hooks/excerpt_allowed_wrapper_blocks/>) which is used by the [`excerpt_remove_blocks` function](<https://developer.wordpress.org/reference/functions/excerpt_remove_blocks/>)
I might be over simplifying due to ignorance, but the term description field doesn'tappear to be far removed from the post excerpt field.
(Ignoring the stack of excerpt generative and santization filter requirements)
My point. The Wordpress Classic Editor block allows for an entity to be edited in both classic and editor modes and the content and assiciated functions survive both experiences. 
Though as I am composing this post it occurs to me that both the excerpt and core/comment blocks require the [block-editor](<https://developer.wordpress.org/block-editor/how-to-guides/platform/custom-block-editor/>) to function, and the form fields required for settings/options would be the same form fields which are still in experimental development phase. For some reason I was dreaming of an opt-in switch that would magically interchange admin settings page with block editor page. (aka wordpress-classic editor plugin) - but realilty just slapped me upside-the-head.

**Comment by jonathan-dejong** [comment]
> While I am also desperately waiting for this I have to strongly disagree here. Due to the backwards compatibility commitment of WP nothing ever is temporary. If it takes time that sucks but is still better than getting a half-baked stopgap solution that will only cause much more pain in the long term.

On principle I agree with this BUT consider that this is a 5 year old thread and one of the biggest hurdles in my opinion with WP as an eco system is that we are consistently falling behind due to this mentality. 
We've gone from inventing the web to just trying to constantly catch up with everyone else. As a developer who's built his whole career on WordPress I love and appreciate it and the community so much, but it's also more and more disheartening experiencing this first hand. 
Having to tell a client that it could be years still for "that block editor we have for pages should also be used for rich content on terms" makes absolutely no sense to them 😞 

WooCommerce now has a block editor experience for product pages which are far more complex than a terms edit page.. @goaround have made a decent POC using the block-everywhere plugin. 
We know this is possible, there's a will to make it happen from the community, a desire for it to be a thing from our users, but  even if I had the time to make it I'm still not seeing any evidence that my efforts would not just be in vain anyway due to some abstract "grand admin redesign plan" that's likely years away 🤷‍♂️  

On a constructive note how we decided to "solve" this for our clients is that we've made a "Block areas" post type which is just a non-public post type where one can create posts containing blocks. Then in edit screens for any terms or other locatiosn where blocks are currently not supported we have two inputs "Add block areas above content" and "Add block areas below content". which allows for searching and picking any number of these posts in both locations. 
For slightly easier admin we've also made sure to link through to the posts and the post table screen from these sections and have added a column to the block area posts table showing where the posts have been added to. 
This is definitely not ideal, but it gives a lot more flexibility to our clients with hardly any error-prone coding like matching up slugs between terms/posts and other solutions posted here. Not that I consider those approaches bad! Just a different way to skin this cat.

**Comment by bph** [comment]
Also suggested on this [Reddit comment:](https://www.reddit.com/r/Wordpress/comments/1fqxhu6/comment/lpbcy7z/) 

> Gutenberg still is not supported for categories, tags, and other term pages, and this issue is at least five[ years old](https://github.com/WordPress/gutenberg/issues/17099). In other words, you have to use stupid workarounds to have some custom content on the category page.
> 
> It's not an issue if you are making some simple marketing sites, but if you have something more complex than a blog, you have to use custom taxonomies and post types.

**Comment by rpldom** [comment]
I really don't understand why team spending time on changing colors and moving fonts here and there in Gutenberg. Also making full block themes in parallel when taxonomy system is still in Netscape era.
Isn't it one of most crucial parts of modern CMS?

**Comment by asafm7** [comment]
I feel that this thread isn't reaching the right people. I wonder who they are.

**Comment by bph** [comment]
@mtias @youknowriad Could this be part of WP redesign project? It seems to be a adoption blocker to not be able to add blocks to Category pages and still have to deal with a basic textarea box.

**Comment by Wohlfarth** [comment]
Obviously this thread hasn't reached the right people.
I have submitted a new feature request at https://core.trac.wordpress.org/ticket/62649
Hopefully someone will react there.
Happy day all!

**Comment by Mamaduka** [comment]
Closing this as "maybe later". This is going to be part of the WP dashboard redesign process. Currently, there's no isolated plan for the taxonomies editor.

**Comment by WesCook** [comment]
I've seen it mentioned a number of times now that this feature needs to be tied to a larger WP dashboard redesign.  Can you explain why that is, @Mamaduka?  Is there a larger refactor happening in how post or category data is stored alongside a UI refresh?

Am happy to read into any deeper discussions, if they exist online.  It just seems strange that this would be a blocking requirement.

**Comment by djcowan** [comment]
> … would be a blocking requirement.

Initial thoughts, 
1. the block editor runs in an iframe and loads assets accordingly.
2. Terms and term content use a different Query 
3. Term content is stored, filtered and sanitised differently

**Comment by Mamaduka** [comment]
@WesCook, here are some make block posts about the admin redesign.

* https://make.wordpress.org/core/2023/07/12/admin-design/
* https://make.wordpress.org/design/2023/08/10/admin-design-kickoff/

> Is there a larger refactor happening in how post or category data is stored alongside a UI refresh?

The data will be consumed via REST API. I don't know if there are plans to change anything in the DB schemas.

**Comment by jonathan-dejong** [comment]
> > … would be a blocking requirement.
> 
> Initial thoughts,
> 
> 1. the block editor runs in an iframe and loads assets accordingly.
> 2. Terms and term content use a different Query
> 3. Term content is stored, filtered and sanitised differently

Not sure what you intend by this? To me this does not in any way explain why it would by blocking? The Block editor talks to the DB through the REST API and there's endpoints for categories just as there are for post types. 
If we were concerned with backwards compatibility that's been discussed here already with proposed solutions. 
IMO adding a new column to the wp_terms table for the block content OR using the existing description would work equally fine with different pros/cons.

> [@WesCook](https://github.com/WesCook), here are some make block posts about the admin redesign.
> 
> * https://make.wordpress.org/core/2023/07/12/admin-design/
> * https://make.wordpress.org/design/2023/08/10/admin-design-kickoff/
> 
> > Is there a larger refactor happening in how post or category data is stored alongside a UI refresh?
> 
> The data will be consumed via REST API. I don't know if there are plans to change anything in the DB schemas.

This further confirms that it would happen through the REST API, so why is it blocking? It seems a natural progression to me to enable it for terms as an early step of the admin redesign effort. 
It's honestly also not instilling a whole lot of confidence and trust when the latest information we have on this supposed "Large admin redesign" are 2 years old blog posts which at "best" shows what we are going to get and at "worst" are just early concept sketches. 🤷  

One lesson from FSE that we as an agency (and I do not believe we were alone) learned was that people don't like drastic changes to their administrative experience. So why not approach the admin redesign as an iterative thing where adding the block editor to terms would serve that purpose while also giving tangible and very real value that is hard to dispute. 

That's my two cents right now :)

**Comment by asafm7** [comment]
I feel like the Gutenberg team is making a consistent mistake here. It's an old and well-defined issue. The ongoing discussion about it proves that it's a real concern.

Regarding resources — over the past few years, many have offered to help, and if I remember correctly, there are already some non-standard solutions ready for implementation.

It feels like the Gutenberg team is quick to dismiss the community here and prefers to stick to unclear, tentative roadmaps.

Of course, I might be missing something.

**Comment by paaljoachim** [comment]
One good way to move this forward is that someone creates a plugin. 
As the plugin could at a later time be incorporated in some form into Gutenberg.

**Comment by asafm7** [comment]
@paaljoachim, I might be wrong because it's been a while since I dealt with it (as I sort of gave up), but I believe a plugin already exists: https://github.com/Automattic/blocks-everywhere

The problem is that, if I remember correctly, it relies on the Gutenberg plugin, which is Beta by definition, and many, understandably, don't want to use it in production.

Also, the Blocks Everywhere hasn't been updated for over a year.

**Comment by jonathan-dejong** [comment]
> One good way to move this forward is that someone creates a plugin. As the plugin could at a later time be incorporated in some form into Gutenberg.

This has been said so many times now it's a statement that has lost all meaning to me. 
Also, to build on @asafm7 there is a plugin for active beta gutenberg development and it's the Gutenberg plugin. 
That is what we're asking for here, to develop this in the plugin (to eventually go into core) because we are in fact IN the Gutenberg issues section :) 

Take this xkcd comic and swap "standard" with plugin for core development and that's how to summarise the feeling I get when someone says that. 
https://xkcd.com/927/

What we're actually saying here is to make a plugin to extend a plugin to extend core. 
Which is what many of the people above (and others on the web) have done and what @asafm7  refers to. Yet we are nowhere closer to actually having this functionality 6 years later.

**Comment by asafm7** [comment]
The WordPress and Gutenberg teams might benefit from using [feature request management software](https://canny.io/use-cases/feature-request-management) to better prioritize tasks and stay in touch with the community. It often feels like things fall between the cracks and are never picked up. Even making it into the Gutenberg plugin doesn’t guarantee progress — the Table of Contents block has been stuck there for years.

It will also enable less technically oriented users to make their voices heard.
